### ISSUE TITLE
Popover floats through the top of the screen

### ISSUE THREAD

**Issue by KashMoneyMillionaire** [vmware/clarity#1213]
**Select one ...**  (check one with "x")
```
[x] bug
[ ] feature request
[ ] enhancement
```

### Expected behavior
Popover modules should not go through the top of the screen when opened

### Actual behavior
Popover goes through the top of the screen and the top of the popover is unreachable

### Reproduction of behavior
https://plnkr.co/OITGa8m3JOVOSexl8KIw

Click the show/hide columns on the datagrid to view the popover

### Environment details

* **Angular version:** 4.3.1

* **Clarity version:** 0.9.14

* **OS and version:** Windows 10

* **Browser:** Chrome, but I assume others as well

### Suggested Fix
When I add the following above Line 244, it prevents it from going off screen. https://github.com/vmware/clarity/blob/e6da210d0c2ab8b06b463278eb729a0d04c00674/src/clarity-angular/popover/common/popover.ts#L244 
```typescript
leftDiff = Math.max(0, leftDiff);
topDiff = Math.max(0, topDiff);
```

**Comment by mathisscott** [comment]
@KashMoneyMillionaire 
I don't see a popover on that plunkr. Was the right plunkr link shared in the description?

**Comment by mathisscott** [comment]
See #1129

**Comment by vinayakpatil** [comment]
Any updates on this one?

**Comment by mathisscott** [comment]
@vinayakpatil 
This is not one of our top three priorities at this time. If the fix described above meets our architectural guidelines, then we could open it up for external contributors. 

@youdz would be the person to answer that question.

**Comment by youdz** [comment]
I see a couple things:
First, it would need to be just a bit more developed than that, both because the position is relative to the closest positioned parent (not the window) and because we'd want it to prevent from going off screen on all sides, not just top and left.
Second, we would need UX to check if this is the behavior they want, because it produces some strange and sometimes frustrating behaviors. For instance, the popover would cover the popover trigger itself, which means clicking on the trigger a second time to close the popover is not possible.

The reason we haven't touched "collision detection" yet is because it's both more complicated than it seems for a library, and because it requires UX work to handle every single edge case.

**Comment by marvinosswald** [comment]
Any update here ? this seems like a show stop to me right now..

**Comment by mathisscott** [comment]
@marvinosswald 
We don't have this scheduled to be implemented in the near term. It's still in early stages at the moment. Are there any details you could share as to why this would be a show stopper for you?

**Comment by marvinosswald** [comment]
@mathisscott while i‘m not absolutely sure this is the right issue maybe #1262. but my issue is that wih a datagrid and multiple columns the first filter glichtes outside the window and is therby not usable.. that seems like quite an issue to me, but i understand why its not a top prio issue as it only occurs if your rows are getting really narrow due to alot of them combined with wide custom filters.

For now i‘m using a css hack to solve this by basicaly disabling ˋtransform-xˋ and reseting ˋleftˋ for the first few columns

**Comment by gnomeontherun** [comment]
This should be resolved as of 2.2.0, closing as fixed by #2923
